src/main/java/org/sonarsource/sonarlint/ls/Utils.java

```diff
@@ -62,4 +65,21 @@ public static void interrupted(InterruptedException e) {
     LOG.debug("Interrupted!", e);
     Thread.currentThread().interrupt();
   }
+
+  public static String encodeUriComponent(String toEncode) {
+    try {
+      return URLEncoder.encode(toEncode, StandardCharsets.UTF_8.name());
+    } catch (UnsupportedEncodingException e) {
+      // Should not happen on a standard JVM
+      throw new IllegalStateException(e);
+    }
+  }
```

Comment:
We already have StringUtils.urlEncode in SLCORE

Reply:
Good catch :+1: